commons-mongo/src/main/scala/com/avsystem/commons/mongo/core/ops/Filtering.scala

```diff
@@ -2,11 +2,20 @@ package com.avsystem.commons
 package mongo.core.ops
 
 import com.avsystem.commons.mongo.{BsonRef, DocKey}
+import com.avsystem.commons.serialization.GenCodec
+import com.mongodb.client.model.Filters
 import org.bson.BsonValue
 import org.bson.conversions.Bson
 
-object Filtering {
+object Filtering extends LowPrioFiltering {
   implicit def bsonFiltering(bson: Bson): BsonFiltering = new BsonFiltering(bson)
   implicit def docKeyFiltering[T](docKey: DocKey[T, _ <: BsonValue]): DocKeyFiltering[T] = new DocKeyFiltering(docKey)
-  implicit def bsonRefFiltering[T](bsonRef: BsonRef[T]): BsonRefFiltering[T] = new BsonRefFiltering(bsonRef)
+  implicit def bsonRefIterableFiltering[E: GenCodec, C[T] <: Iterable[T]](bsonRef: BsonRef[C[E]]): BsonRefIterableFiltering[E, C] = {
```

Comment:
Are you fine with the fact that this will also catch `Map`s? They don't serialize to arrays so this may fail completely silently in runtime (just a query returning empty result without any error or warning).